packages/neon/neon_talk/lib/src/widgets/message.dart

```diff
@@ -89,3 +92,274 @@ class TalkMessagePreview extends StatelessWidget {
     );
   }
 }
+
+/// Displays a Talk chat message in the right way based on the message type.
+class TalkMessage extends StatelessWidget {
+  /// Creates a new Talk message.
+  const TalkMessage({
+    required this.chatMessage,
+    this.previousChatMessage,
+    this.isParent = false,
+    super.key,
+  });
+
+  /// {@template TalkMessage.chatMessage}
+  /// The chat message to display.
+  /// {@endtemplate}
+  final spreed.$ChatMessageInterface chatMessage;
+
+  /// {@template TalkMessage.previousChatMessage}
+  /// The previous chat message which must not be a system message.
+  /// {@endtemplate}
+  final spreed.$ChatMessageInterface? previousChatMessage;
+
+  /// {@template TalkMessage.isParent}
+  /// Whether the displayed chat message is a parent chat message that was replied to.
+  /// {@endtemplate}
+  final bool isParent;
+
+  @override
+  Widget build(BuildContext context) {
+    assert(
+      previousChatMessage?.messageType != spreed.MessageType.system,
+      'previousChatMessage most not be a system message',
+    );
+
+    if (chatMessage.messageType == spreed.MessageType.system) {
+      return TalkSystemMessage(
+        chatMessage: chatMessage,
+      );
+    }
+
+    return TalkCommentMessage(
+      chatMessage: chatMessage,
+      previousChatMessage: previousChatMessage,
+      isParent: isParent,
+    );
+  }
+}
+
+/// Displays a system chat message.
+class TalkSystemMessage extends StatelessWidget {
+  /// Creates a new Talk system message.
+  const TalkSystemMessage({
+    required this.chatMessage,
+    super.key,
+  });
+
+  /// {@macro TalkMessage.chatMessage}
+  final spreed.$ChatMessageInterface chatMessage;
+
+  @override
+  Widget build(BuildContext context) {
+    if (chatMessage.systemMessage == 'reaction' ||
+        chatMessage.systemMessage == 'reaction_revoked' ||
+        chatMessage.systemMessage == 'reaction_deleted' ||
+        chatMessage.systemMessage == 'message_deleted') {
+      return const SizedBox.shrink();
+    }
```

Comment:
add all strings into a constant set and check whether it contains the current message